modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java

```diff
@@ -662,8 +644,15 @@ public void add(FeatureEntry entry, SimpleFeatureCollection collection) throws I
                     SimpleFeature f = it.next();
                     SimpleFeature g = w.next();
                     for (PropertyDescriptor pd : collection.getSchema().getDescriptors()) {
+                        /* geopkg spec requires booleans to be stored as SQLite integers this fixes
+                         * bug reported by https://osgeo-org.atlassian.net/browse/GEOT-5904 */
                         String name = pd.getName().getLocalPart();
-                        g.setAttribute(name, f.getAttribute(name));
+                        if (pd.getType().getBinding() == Boolean.class) {
+                            int bool = (Boolean)(f.getAttribute(name)) ? 1 : 0;
```

Comment:
Few questions about this change:
1. What happens if the value is null? You probably need a null check here.
2. What happens if you get a list of Features from a GeoPackage store? Will they be converted back to Booleans?
3. Is it possible for there to be GeoPackage files written by GeoTools before this fix that contain boolean values, and does GeoTools still read these correctly?
4. Does this affect the GeoServer GeoPackage output format; does this bug show up there?

Reply:
1. Null value check added and tested.
2.Values from a geopackage, according to the spec, will already be of type Integer and will remain that way when being written to another geopackage.
3. While it was possible for GeoTools to write GeoPackages containing Boolean fields, the data in those fields would have been Integers and not Strings.  This fix allows GeoTools to convert a Boolean in String form either 'T' 'True' 'true' etc.. to 1 or 0 as integers.
4. This bug was found by exporting a Layer that was originally stored as a shapefile as a geopackage.  In tracing the process, the problem was discovered in the GeoTools GeoPackage.java class where the fix now resides.  This fix as been tested with a running instance of GeoServer and proven to resolve the issue.